numpy/core/src/_simd/_simd_inc_convert.h

```diff
@@ -41,39 +41,38 @@ simd_scalar_to_obj(simd_data data, simd_data_type dtype)
     return PyLong_FromUnsignedLongLong(data.u64 >> leftb);
 }
 
+typedef struct {
+    size_t len;
+    void *ptr;
+} simd__alloc_data;
+
 static void *
 simd_sequence_new(Py_ssize_t len, simd_data_type dtype)
 {
     const simd_data_info *info = simd_data_getinfo(dtype);
     assert(info->is_sequence && info->lane_size > 0);
-
-    size_t size  = NPY_SIMD_WIDTH + sizeof(size_t) + sizeof(size_t*);
-           size += len * info->lane_size;
-
-    size_t *ptr = malloc(size);
+    size_t size = sizeof(simd__alloc_data) + len * info->lane_size + NPY_SIMD_WIDTH;
+    void *ptr = malloc(size);
     if (ptr == NULL) {
         return PyErr_NoMemory();
     }
-    *(ptr++) = len;
-    size_t **a_ptr = (size_t**)(
-        ((size_t)ptr + NPY_SIMD_WIDTH) & ~(size_t)(NPY_SIMD_WIDTH-1)
+    // align the pointer
+    simd__alloc_data *a_ptr = (simd__alloc_data *)(
+        ((uintptr_t)ptr + sizeof(simd__alloc_data) + NPY_SIMD_WIDTH) & ~(uintptr_t)(NPY_SIMD_WIDTH-1)
     );
-    a_ptr[-1] = ptr;
+    a_ptr[-1].len = len;
+    a_ptr[-1].ptr = ptr;
     return a_ptr;
 }
-
 static size_t
```

Comment:
```suggestion
static Py_ssize_t 
```
since the `len` argument to `simd_sequence_new` is also of this type.

Reply:
wait `simd_sequence_new()` should only accept unsigned integer, using `Py_ssize_t` seems weird.
should I use `size_t` with `simd_sequence_new()` or `unsigned int` to avoid overflow?

Reply:
`unsigned int` or just `unsigned` would be friendly when casting to `Py_ssize_t`.

Reply:
I would stick with `Py_ssize_t` in `simd_sequence_new` (to match `PyList_New`), but throw `ValueError` if `len <= 0`.

Is `len == 0` legal anyway?

Reply:
`Py_ssize_t` is now used instead of `size_t` and `unsigned`, also I left `assert` case for `len`.